### PR TITLE
[ZEPPELIN-474] Provide InterpreterContext as ThreadLocal value

### PR DESCRIPTION
**Pull request by Leemoonsoo**
This PR implements ZEPPELIN-474.

Let 3rd party library access InterpreterContext.

### CODE REVIEW

File: zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterContext.java [hunk 1]
```diff
@@ -27,6 +27,21 @@
  * Interpreter context
  */
 public class InterpreterContext {
+  private static final ThreadLocal<InterpreterContext> threadIC =
```
Comment (bzz) [inline zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterContext.java]: Would it be too big change to make `InterpreterContext` a singleton, and then add non-static field of the `TreadLocal`?

**Comment by bzz** [comment]
Looks good to me

File: zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterContext.java (hunk 1 above)
Comment (Leemoonsoo) [inline zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterContext.java]: I'm not sure if i understand your suggestion correctly, but InterpreterContext is created every time. For example, new instance of InterpreterContext is created passed into Interpreter.interpret() method. So i think it's hard to make InterpreterContext a singleton.

Comment (bzz) [inline zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterContext.java]: You are right, thank your for the explanation!

**Comment by Leemoonsoo** [comment]
I'm merging it into master

**Pull request closed without merging**